src/plots/smith/smith.js

```diff
@@ -0,0 +1,1456 @@
+
+    var bboxIndices = {
+        x: [0, 2],
+        y: [1, 3]
+    };
+
+    ax.setRange = function() {
+        var sectorBBox = _this.sectorBBox;
+        var ind = bboxIndices[axId];
+        var rl = _this.radialAxis._rl;
+        var drl = (rl[1] - rl[0]) / (1 - smithLayout.hole);
+        ax.range = [sectorBBox[ind[0]] * drl, sectorBBox[ind[1]] * drl];
+    };
+
+    ax.isPtWithinRange = axId === 'x' ?
+        function(d) { return _this.isPtInside(d); } :
+        function() { return true; };
+
+    ax.setRange();
+    ax.setScale();
+    return ax;
+};
+
+proto.doAutoRange = function(fullLayout, smithLayout) {
+    var gd = this.gd;
+    var radialAxis = this.radialAxis;
+    var radialLayout = smithLayout.realaxis;
+
+    radialAxis.setScale();
+    doAutoRange(gd, radialAxis);
+
+    var rng = radialAxis.range;
+    radialLayout.range = rng.slice();
+    radialLayout._input.range = rng.slice();
+
+    radialAxis._rl = [
+        radialAxis.r2l(rng[0], null, 'gregorian'),
+        radialAxis.r2l(rng[1], null, 'gregorian')
+    ];
+};
+
+proto.updateRadialAxis = function(fullLayout, smithLayout) {
+    var _this = this;
+    var gd = _this.gd;
+    var layers = _this.layers;
+    var radius = _this.radius;
+    var cx = _this.cx;
+    var cy = _this.cy;
+    var radialLayout = smithLayout.realaxis;
+    var a0 = mod(smithLayout.sector[0], 360);
+    var ax = _this.radialAxis;
+    var hasRoomForIt = true;
+
+    _this.fillViewInitialKey('realaxis.angle', radialLayout.angle);
+    _this.fillViewInitialKey('realaxis.range', ax.range.slice());
+
+    ax.setGeometry();
+
+    // rotate auto tick labels by 180 if in quadrant II and III to make them
+    // readable from left-to-right
+    //
+    // TODO try moving deeper in Axes.drawLabels for better results?
+    if(ax.tickangle === 'auto' && (a0 > 90 && a0 <= 270)) {
+        ax.tickangle = 180;
+    }
+
+    var transFn2 = function(d) {
+        return strTranslate(ax.c2p(d.x), -2);
+    };
+
+    // set special grid path function
+    var gridPathFn = function(d) {
+        var value = d.x;
+
+        var gridRadius = 0.5 * (_this.radius - ax.c2p(value));
+        var gridCenter = gridRadius + ax.c2p(value);
+        return Lib.pathArc(gridRadius, 0, 2 * Math.PI, gridCenter, 0);
+    };
+
+    var newTickLayout = strTickLayout(radialLayout);
+    if(_this.radialTickLayout !== newTickLayout) {
+        layers['radial-axis'].selectAll('.xtick').remove();
+        _this.radialTickLayout = newTickLayout;
+    }
+
+    var labelFns = {};
+
+    labelFns.xFn = function() {
+        return 0;
+    };
+
+    labelFns.yFn = function() {
+        return 0;
+    };
+
+    labelFns.anchorFn = function() {
+        return 'end';
+    };
+
+    labelFns.heightFn = function() {
+        return 0;
+    };
+
+    if(hasRoomForIt) {
+        ax.setScale();
+
+        // circular grid lines
+        Axes.drawGrid(gd, ax, {
+            vals: [{x: 0.2 }, {x: 0.5 }, {x: 1.0 }, {x: 2.0 }, {x: 5.0 } ],
```

Comment:
Let's declare this constant once.

Reply:
cb07c28f397c2b98f059b47fbacf5870039f15c0